app/components/timer_session_entry_component.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
```

Comment:
If you don't want to continue the discussion about view components you can just resolve this comment :)

But If you do... I dislike view components because:
* It separates the template specific assertions from the template.
* It creates boilerplate. In this example 26 Lines (30 - 4). 4 because the function bodies are the only relevant code.
* They achieve something in 2 files which could be done 1. (Makes it harder to navigate)
* It doesn't reduce the amount of code in the template.
* It takes me longer to understand and change a view component than a normal template because of the before mentioned reasons.
* They don't make the template dumber as they don't reduce `if`-statements or anything else.
* They live in a "shared" but often are just used a single time.

Reply:
in shared? 😆 they don't life in shared

Reply:
> They don't make the template dumber as they don't reduce if-statements or anything else.

They ensure that if statements are tested.. by moving them to a ruby class you expose them to code-cov

Reply:
> It separates the template specific assertions from the template.

Which is good as there should be no template specific assertions :D if we would write better code there would only be data to be displayed .. not data to be switched on

Reply:
* `app/components` is a sort of shared folder even if it's not called "shared".
* There are no unit test tests for this class. You probably achieved the coverage with a system test so it's the equivalent of a test called: "It compiles" or did you explicitly test the behaviour of this if statement? (which I would not do in a system test)
* I don't think the "good code" you are talking about is very particle.